addons/product/models/product_template.py

```diff
@@ -451,8 +466,11 @@ def _onchange_uom(self):
 
     @api.onchange('type')
     def _onchange_type(self):
-        # Do nothing but needed for inheritance
-        return {}
+        if self.type == 'combo':
+            if self.attribute_line_ids:
+                raise UserError(_("Combo products can't have attributes"))
+            self.taxes_id = False
+            self.supplier_taxes_id = False
```

Comment:
I just saw this while reviewing quickly while on leave, but these fields are declared in `account` :woozy_face:

Reply:
Good catch! I'll move part of the `onchange` to `account` :)